react/Popup/index.jsx

```diff
@@ -0,0 +1,133 @@
+import { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+/**
+ * customized function to center a popup window
+ * @param  {string} url
+ * @param  {string} title
+ * @param  {string|number} w
+ * @param  {string|number} h
+ * @return {Window}       Popup window
+ */
+// source https://stackoverflow.com/a/16861050
+export function popupCenter(url, title, w, h) {
+  /* global screen */
+  // Fixes dual-screen position
+  //                      Most browsers      Firefox
+  var dualScreenLeft = window.screenLeft || screen.left
+  var dualScreenTop = window.screenTop || screen.top
+
+  var width = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+    ? document.documentElement.clientWidth
+    : screen.width
+  var height = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+    ? document.documentElement.clientHeight
+    : screen.height
+
+  var left = width / 2 - w / 2 + dualScreenLeft
+  var top = height / 2 - h / 2 + dualScreenTop
+  // need to be set here to get from the OAuth opener
```

Comment:
I think that comment has lost it's context? Otherwise what's the OAuth opener here?